### ISSUE TITLE
Fluent bit v0.14.6 crash with SEGFAULT when Grep plugin is used followed by Modify_record or Lua

### ISSUE THREAD

**Issue by nikinaydenov** [fluent/fluent-bit#892]
## Bug Report

**Describe the bug**
Fluent bit crash with segmentation fault since v0.14.6 when you use  the "grep" filter with exclude.
    
**To Reproduce**
- Rubular link if applicable:
- Example log message if applicable:
```
{"log": "2018/11/07 09:47:45 I! Using config file: /etc/telegraf/telegraf.conf"}
```
- Steps to reproduce the problem:
The setup I use is docker container (in the example it is based on telegraf image). I just restart the container. I exclude like:
[FILTER]
    Name grep
    Match hubtelegraf
    Exclude log Using config file:

When I restart the container the fluent bit crash.

Another easy way to reproduce the problem (no crash but visible unexpected behavior):
cat test.sh
#!/bin/sh
while :; do
  echo -n '{"container_id": "7cebceb5dc7baa19629b9321f6da6cf0a65c4f3f5faf3d3fbf41b392a6f82102", "container_name": "/awsbaseimage_telegraf_1", "source": "stderr", "log": "2018/11/07 08:07:16 I! Using config file: /etc/telegraf/telegraf.conf"}
  sleep 5
done

Config:
[SERVICE]
    Flush        5
    Daemon       Off
    Log_Level    info
    Parsers_File parsers.conf
    HTTP_Server  Off

[INPUT]
    Name        stdin
    Tag         hubtelegraf

[INPUT]
    Name        forward
    Listen      127.0.0.1
    Port        24224
    Chunk_Size  32
    Buffer_Size 64

[FILTER]
    Name grep
    Match hubtelegraf
    Exclude log Using config file:

[FILTER]
    Name     parser
    Match    hubtelegraf
    Key_Name log
    Parser   hubtelegraf
    Reserve_Data true

[FILTER]
    Name   record_modifier
    Match  hubtelegraf
    Record service telegraf

[FILTER]
    Name record_modifier
    Match hub*
    Remove_key container_id
    Remove_key container_name

[FILTER]
    Name lua
    Match *
    script remove_color.lua
    call rmcolor

[OUTPUT]
    Name stdout
    Match *

Parser is:
[PARSER]
    Name    hubtelegraf
    Format  regex
    Regex   ^(?<timestamp>\d\d\d\d-\d\d-\d\dT\d\d:\d\d:\d\dZ) (?<loglevel>.)! (?<message>.*)$
    Time_Key    timestamp
    Time_Format %Y-%m-%dT%H:%M:%S
    Time_Keep   Off

I want to exclude this line because it does not match timestamp (the only line in the whole log stream that do not match and I am not interested in it). Here in the example I just send it on each 5 sec to the stdin plugin.

Run it as:
/opt/td-agent-bit.0.14.6/bin/test.sh|/opt/td-agent-bit.0.14.6/bin/td-agent-bit -c /etc/td-agent-bit/fluentbit-crash.conf

And you will receive:
Fluent-Bit v0.14.6
Copyright (C) Treasure Data

[2018/11/07 10:03:32] [ info] [engine] started (pid=10099)
[2018/11/07 10:03:32] [ info] [in_fw] binding 127.0.0.1:24224
[2018/11/07 10:03:36] [error] [src/flb_input.c:927 errno=12] Cannot allocate memory
[2018/11/07 10:03:41] [error] [src/flb_input.c:927 errno=12] Cannot allocate memory

**Expected behavior**
Up to v.0.14.5 it is working and  the next line is silently excluded
/opt/td-agent-bit.0.14.6/bin/test.sh|/opt/td-agent-bit/bin/td-agent-bit -c /etc/td-agent-bit/fluentbit-crash.conf
Fluent-Bit v0.14.5
Copyright (C) Treasure Data

[2018/11/07 10:04:29] [ info] [engine] started (pid=10190)
[2018/11/07 10:04:29] [ info] [in_fw] binding 127.0.0.1:24224

And no more output. 

**Screenshots**

....
[135103] hubtelegraf: [1541583858.000000000, #<UNKNOWN 135104 140164408197168>]
[135104] hubtelegraf: [1541583858.000000000, #<UNKNOWN 135105 140164408197168>]
[135105] hubtelegraf: [1541583858.000000000, #<UNKNOWN 135106 140164408197168>]
[135106] hubtelegraf: [1541583858.000000000, #<UNKNOWN 135107 140164408197168>]
[engine] caught signal (SIGSEGV)
[2018/11/07 09:44:30] [ warn] unknown time format 2ab2a
[2018/11/07 09:44:30] [ warn] unknown time format 2ab2b
[2018/11/07 09:44:30] [ warn] unknown time format 2ab2c
[2018/11/07 09:44:30] [ warn] unknown time format 2ab2d
[2018/11/07 09:44:30] [ warn] unknown time format 2ab2e
[2018/11/07 09:44:30] [ warn] unknown time format 2ab2f
[2018/11/07 09:44:30] [ warn] unknown time format 2ab30
[2018/11/07 09:44:30] [ warn] unknown time format 2ab31
[2018/11/07 09:44:30] [ warn] unknown time format 2ab32
[2018/11/07 09:44:30] [ warn] unknown time format 2ab33
[2018/11/07 09:44:30] [ warn] unknown time format 2ab34
[2018/11/07 09:44:30] [ warn] unknown time format 2ab35
[2018/11/07 09:44:30] [ warn] unknown time format 2ab36
[2018/11/07 09:44:30] [ warn] unknown time format 2ab37
[2018/11/07 09:44:30] [ warn] unknown time format 2ab38
[2018/11/07 09:44:30] [ warn] unknown time format 2ab39
[2018/11/07 09:44:30] [ warn] unknown time format 2ab3a
[2018/11/07 09:44:30] [ warn] unknown time format 2ab3b
[2018/11/07 09:44:30] [ warn] unknown time format 2ab3c
[2018/11/07 09:44:30] [ warn] unknown time format 2ab3d
[2018/11/07 09:44:30] [ warn] unknown time format 2ab3e
[2018/11/07 09:44:30] [ warn] unknown time format 2ab3f
[2018/11/07 09:44:30] [ warn] unknown time format 2ab40
[2018/11/07 09:44:30] [ warn] unknown time format 6
[2018/11/07 09:44:30] [ warn] unknown time format 7f7a91d0
[2018/11/07 09:44:30] [ warn] unknown time format 2ab95
[2018/11/07 09:44:30] [ warn] unknown time format 2ab96
[2018/11/07 09:44:30] [ warn] unknown time format 2ab97
[2018/11/07 09:44:30] [ warn] unknown time format 2ab98
ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)ERROR: no debug info in ELF executable (-1)[135107] hubtelegraf: [Aborted (core dumped)

**Your Environment**

* Version used: 0.14.6
* Configuration: See above
* Environment name and version (e.g. Kubernetes? What version?): docker compose 
* Server type and version: ec2 in AWS
* Operating System and version: Ubuntu 16.04
* Filters and plugins: Grep, Parcer, Record_modifier, Lua, but in fact it is any of Grep + Record_modifier or Grep + Lua that reproduce the problem.

**Additional context**

I am not able to use the Grep plugin, as it kills the fluent bit with Segmentation Fault (when used with forward input plugin). When problem is demonstrated with only one line to stdin and this line is excluded by grep plugin you see the above error. When this line is a part of log stream coming from the docker container over the forward plugin fluent bit crash on the next line after the excluded one.  I can provide a coredump.

**Comment by edsiper** [comment]
@nikinaydenov thanks for reporting this issue, the bug has been fixed and the fix will be available in the next release.

**Comment by edsiper** [comment]
Fixed by f8b99792
